addons/base_vat/base_vat.py

```diff
@@ -328,3 +334,33 @@ def check_vat_tr(self, vat):
             return int(vat[9]) == c1 and int(vat[10]) == c2
 
         return False
+
+    def get_address(self, cr, uid, vat, context=None):
+        vals = {}
+        user_company = self.pool['res.users'].browse(cr, uid, uid, context=context).company_id
+        if user_company.vat_check_vies:
+            try:
+                vat_country, vat_number = self._split_vat(vat)
+                client = Client(vatnumber.VIES_URL)
+                response = client.service.checkVat(countryCode=vat_country, vatNumber=vat_number)
+                if response['valid']:
+                    country_id = self.pool['res.country'].search(cr, uid, [('code', '=ilike', vat_country)], context=context)
```

Comment:
Actually country code is unique so we get only one record, anyway i have puted limit.